include/internal/catch_test_registry.h

```diff
@@ -101,5 +109,84 @@ struct AutoReg : NonCopyable {
         Catch::AutoReg INTERNAL_CATCH_UNIQUE_NAME( autoRegistrar )( Catch::makeTestInvoker( Function ), CATCH_INTERNAL_LINEINFO, Catch::StringRef(), Catch::NameAndTags{ __VA_ARGS__ } ); /* NOLINT */ \
         CATCH_INTERNAL_UNSUPPRESS_GLOBALS_WARNINGS
 
+    ///////////////////////////////////////////////////////////////////////////////
+    #define INTERNAL_CATCH_TEMPLATE_TEST_CASE_2(TestName, TestFunc, Name, Tags, ... )\
+        CATCH_INTERNAL_SUPPRESS_GLOBALS_WARNINGS \
+        template<typename TestType> \
+        static void TestFunc();\
+        template<typename Type, typename...Rest> \
+        struct TestName : TestName<Rest...> {\
+            template<typename T, typename...Txts> \
+            TestName(T name, Txts...names) : TestName<Rest...>(names...){\
+                Catch::AutoReg INTERNAL_CATCH_UNIQUE_NAME( autoRegistrar )( Catch::makeTestInvoker( &TestFunc<Type> ), CATCH_INTERNAL_LINEINFO, Catch::StringRef(), Catch::NameAndTags{ name, Tags } ); /* NOLINT */ \
+            }\
+        };\
+        template<typename Type> \
+        struct TestName<Type> {\
+            template<typename T>\
+            TestName(T name) {\
+                Catch::AutoReg INTERNAL_CATCH_UNIQUE_NAME( autoRegistrar )( Catch::makeTestInvoker( &TestFunc<Type> ), CATCH_INTERNAL_LINEINFO, Catch::StringRef(), Catch::NameAndTags{ name, Tags } ); /* NOLINT */ \
+            }\
+        };\
+        INTERNAL_CATCH_TEMPLATE_REGISTRY_INITIATE(TestName, Name, __VA_ARGS__) \
```

Comment:
The above should be enclosed in an anonymous namespace to decrease linker pressure.